config/config.go

```diff
@@ -222,3 +241,27 @@ func BuildConfig() (*Config, error) {
 	return c, nil
 	// NOTE: what happens if there no config file and also no env vars setted?
 }
+
+// getVcsRevision gets the buildInfo vcs revision if available.
+func getVcsRevision() string {
+	if x, ok := debug.ReadBuildInfo(); ok {
+		rev := ""
+		mod := ""
+		for _, s := range x.Settings {
+			switch s.Key {
+			case "vcs.revision":
+				if len(s.Value) == 40 { // is the commin hash length
+					rev = s.Value[:8]
+				} else {
+					rev = s.Value
+				}
+			case "vcs.modified":
+				if s.Value == "true" {
+					mod = "*"
+				}
+			}
+		}
+		return rev + mod
+	}
+	return ""
+}
```

Comment:
It is better to use early returns. Also, s/commin/commit/.

```suggestion
func getVcsRevision() string {
	buildInfo, ok := debug.ReadBuildInfo()
	if !ok {
		return ""
	}
		
	rev := ""
	mod := ""
	for _, s := range x.Settings {
		switch s.Key {
		case "vcs.revision":
			if len(s.Value) == 40 { // commit hash length
				rev = s.Value[:8]
			} else {
				rev = s.Value
			}
		case "vcs.modified":
			if s.Value == "true" {
				mod = "*"
			}
		}
	}
	return rev + mod
}
```

Reply:
done in 537475f